packages/cozy-harvest-lib/src/components/AccountModal.jsx

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import get from 'lodash/get'
+import { withMutations } from 'cozy-client'
+import { withRouter } from 'react-router'
+import Spinner from 'cozy-ui/transpiled/react/Spinner'
+import { ModalContent } from 'cozy-ui/transpiled/react/Modal'
+import Infos from 'cozy-ui/transpiled/react/Infos'
+import Button from 'cozy-ui/transpiled/react/Button'
+import { translate } from 'cozy-ui/transpiled/react/I18n'
+
+import accountMutations from '../connections/accounts'
+import triggersMutations from '../connections/triggers'
+import * as triggersModel from '../helpers/triggers'
+import KonnectorConfiguration from './KonnectorConfiguration/KonnectorConfiguration'
+import AccountSelectBox from './AccountSelectBox/AccountSelectBox'
+import KonnectorModalHeader from './KonnectorModalHeader'
+
+/**
+ * AccountModal take an accountId and a list of accounts containing their
+ * respecting triggers and display the selected account and the accounts linked
+ * to this konnector
+ *
+ * You have to pass an array of accounts containing their associated trigger:
+ * ```
+ * accounts: [
+ *  {
+ *    account: {
+ *       _id: '',
+ *    },
+ *    trigger: {
+ *      id: '',
+ *      current_state: {
+ *      }
+ *    }
+ *  }
+ * ]
+ * ```
+ *
+ * from this array and from the passed accountId, we can fetch the account.
+ *
+ */
+export class AccountModal extends Component {
+  state = {
+    trigger: null,
+    account: null,
+    fetching: true,
+    error: false
+  }
+  async componentDidMount() {
+    await this.loadSelectedAccountId()
+  }
+
+  async componentDidUpdate(prevProps) {
+    if (this.props.accountId !== prevProps.accountId) {
+      return await this.loadSelectedAccountId()
+    }
+  }
+
+  async loadSelectedAccountId() {
+    const { accountId, accounts } = this.props
+    const matchingTrigger = get(
+      accounts.find(account => account.account._id === accountId),
+      'trigger'
+    )
+    if (matchingTrigger) await this.fetchAccount(matchingTrigger)
+  }
+
+  async fetchAccount(trigger) {
+    const { findAccount } = this.props
+    this.setState({ fetching: true })
+
+    try {
+      const account = await findAccount(triggersModel.getAccountId(trigger))
+      this.setState({
+        account,
+        trigger
+      })
+      return account
+    } catch (error) {
+      this.setState({
+        error
+      })
+    } finally {
+      this.setState({
+        fetching: false
+      })
+    }
+  }
+
+  render() {
+    const { konnector, onDismiss, history, accounts, t } = this.props
+    const { trigger, account, fetching, error } = this.state
+    return (
+      <>
+        <KonnectorModalHeader konnector={konnector}>
+          {account && (
+            <AccountSelectBox
+              selectedAccount={account}
+              accountsList={accounts}
```

Comment:
To me the AccountSelectBox should be queryConnected (or via Query) to the list of accounts. This would remove the need of passing the list of accounts and also would diminish the risk of using stale data. I think the main blocker here is to be able not to refetch accounts if they already have been loaded.